Wpb 22590 2 wiab dev (#102)
wireapp/wire-docs: src/how-to/install/wiab-dev.md

# WIAB Dev (Wire-in-a-Box development) Deployment Guide

## Introduction

This guide provides detailed instructions for deploying Wire-in-a-Box (WIAB Dev) using Ansible on an Ubuntu 24.04 system. The deployment corresponds to **WIAB Dev (previously called *WIAB Demo*)** in the [planning overview](planning.md#wiab-dev-single-vm-wire-in-a-box). It is a single-node, Minikube-based environment intended for functional testing and evaluation only.

> This will install the wire-server components including the databases in a dev capacity. This setup is not recommended in production but will get you started. Dev version means no data persistence, as everything is stored in memory and will be lost if restarted. It does not require any external storage solutions to function. 

Reviewer: ```suggestion
This will install the wire-server components including the databases in a fashion ready for development or evaluation. This setup is not recommended for production use, but will get you started with learning the product. This configuration has NO data persistence, as everything is stored in memory and will be lost if restarted. It does not require any external storage solutions to function. 
```

> > Read the section [Cleaning/Uninstalling Wire-in-a-Box](#cleaninguninstalling-wire-in-a-box) to clean the installation post testing the dev solution.

Reviewer: ```suggestion
> Read the section [Cleaning/Uninstalling Wire-in-a-Box](#cleaninguninstalling-wire-in-a-box) to clean the installation once your testing or evaluation is complete.
```

- Wire-account-pages, user account management (a few pages relating to e.g. password reset), team-settings page
- Email relay service i.e. smtp service
- Calling services i.e. coturn & SFTD
- Ephemeral datastores
- A cert-manager with `letsencrypt` as an issuer (if `use_cert_manager=true`)

### What will not be installed?

- notifications over native push notifications via [FCM](https://firebase.google.com/docs/cloud-messaging/)/[APNS](https://developer.apple.com/notifications/)
- persistent datastores in k8s 
- high availability

### Relation to WIAB Staging and Production

- **WIAB Staging** runs a similar set of components across multiple VMs and is suitable for more realistic infrastructure validation but without HA. See the [WIAB Staging Deployment](planning.md#wiab-staging-multi-vm-kvm-based-test-cluster) guide.
- **Production** uses dedicated database VMs and a full Kubernetes cluster to provide the maximum capacity and security. See [production installation](planning.md#production-installation-persistent-data-high-availability) for more details.

## Architecture overview

![Wire in a Box Dev Architecture](img/architecture-wiab-dev.png)

> The deployment process is structured into multiple blocks within the Ansible playbooks, offering flexibility in execution. It is designed to configure a remote node, such as example.com (referred to as deploy_node), to install Wire with a custom domain, example.com (referred to as target_domain). These variables must be verified in the file `ansible/inventory/demo/host.yml` (as explained below) before running the pipeline.

Reviewer: ```suggestion
The deployment process is structured into multiple sections within the Ansible playbooks, offering flexibility in execution. It is designed to configure a remote node (referred to as deploy_node), to install Wire with a given domain(referred to as target_domain). These variables must be verified in the file `ansible/inventory/demo/host.yml` (as explained below) before running the pipeline.
```

- **Ansible Playbooks**: 
  - The `ansible` directory from [wire-server-deploy repository](https://github.com/wireapp/wire-server-deploy)
  - Obtain it using **either** method:

>     - **Download as ZIP:** [wire-server-deploy/archive/master.zip](https://github.com/wireapp/wire-server-deploy/archive/refs/heads/master.zip) (requires unzip, wget)

Reviewer: learn about curl. ;)

```

> Note: Optionally the steps `10. Asset Host Setup` and `11. Container Seeding` can also perform these image-extraction and loading steps for you: `setup-offline-sources.yml` will unarchive and host the images via a simple HTTP asset host, and `seed-offline-containerd.yml` will pull/load those images into Minikube node. Those playbooks are tuned for Minikube but can be adapted to work with your own cluster by creating an appropriate inventory and adjusting paths.

### kubeconfig path used by Helm in this deployment

If you are using your own Kubernetes cluster instead of Minikube, ensure that the kubeconfig for your cluster is available at `"/home/{{ ansible_user }}/.kube/config"` before running the `helm_install` step.

### Edit the helm chart values and secrets manually
The playbook generates Helm values and secrets files under `{{ ansible_user_dir }}/wire-server-deploy/values/` (for example `values/wire-server/values.yaml` and `values/wire-server/secrets.yaml`). These files can be edited manually before running the `helm_install` step if you need to change chart values or secrets.

## Trying Things Out post installation

> At this point, with a bit of luck, everything should be working. If not, refer to the [Troubleshooting](#troubleshooting) section below.

Reviewer 2: `with a bit of luck` ;)  What part of the processes requires luck here?

Author: Just a user friendly documentation ;) I templated it out from other pages we have:
https://docs.wire.com/latest/how-to/install/helm-prod.html?h=#trying-things-out

Reviewer 2: This is machine, how luck can help here? When the luck is involved in the business then something we really don't understand or missed to check. My suggestion is either remove this wording if you think everything in place or put the right things to do instead of one being called off as unlucky!
